File: sample/src/main/java/com/schibsted/spain/barista/sample/WrappedEditTextActivity.java
package com.schibsted.spain.barista.sample;

import android.content.Context;
import android.database.Cursor;
import android.database.MatrixCursor;
import android.os.Bundle;
import androidx.annotation.NonNull;
import com.google.android.material.textfield.TextInputLayout;
import androidx.cursoradapter.widget.CursorAdapter;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.SearchView;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

public class WrappedEditTextActivity extends AppCompatActivity {

  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_wrapped_edittext);

    TextView searchResult = findViewById(R.id.searchResult);

    android.widget.SearchView searchView = findViewById(R.id.searchview);
    SearchView supportSearchView = findViewById(R.id.supportSearchView);

    applyListener(searchResult, searchView);
    applySupportListener(searchResult, supportSearchView);

    applyAdapter(searchView);
    applySupportAdapter(supportSearchView);

    TextInputLayout textInputLayout = findViewById(R.id.textInput);
    applyTextInputListener(searchResult, textInputLayout);
  }

  private void applyListener(TextView searchResult, android.widget.SearchView searchView) {
    searchView.setOnQueryTextListener(new android.widget.SearchView.OnQueryTextListener() {
      @Override
      public boolean onQueryTextSubmit(String query) {
        searchResult.setText(query);
        return true;
      }

      @Override
      public boolean onQueryTextChange(String newText) {
        searchResult.setText(newText);
        return true;
      }
    });
  }

  private void applySupportListener(TextView searchResult, SearchView searchView) {
    searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
      @Override
      public boolean onQueryTextSubmit(String query) {
        searchResult.setText(query);
        return true;
      }

      @Override
      public boolean onQueryTextChange(String newText) {
        searchResult.setText(newText);
        return true;
      }
    });
  }

  private void applyTextInputListener(TextView searchResult, TextInputLayout textInputLayout) {
    EditText editText = textInputLayout.getEditText();
    if (editText != null) {
      editText.addTextChangedListener(new TextWatcher() {
        @Override
        public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

        }

        @Override
        public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {
          searchResult.setText(charSequence);
        }

        @Override
        public void afterTextChanged(Editable editable) {

        }
      });
    }
  }

  private void applyAdapter(android.widget.SearchView searchView) {
    MatrixCursor cursor = generateCursor();
    android.widget.CursorAdapter adapter = new CustomAdapter(this, cursor, true);
    searchView.setSuggestionsAdapter(adapter);
  }

  private void applySupportAdapter(SearchView searchView) {
    MatrixCursor cursor = generateCursor();
    CursorAdapter adapter = new CustomSupportAdapter(this, cursor, true);
    searchView.setSuggestionsAdapter(adapter);
  }

  @NonNull
  private MatrixCursor generateCursor() {
    MatrixCursor cursor = new MatrixCursor(new String[] { "_id", "name" });
    cursor.addRow(new Object[] { 0, "a" });
    cursor.addRow(new Object[] { 1, "b" });
    return cursor;
  }

  private static class CustomAdapter extends android.widget.CursorAdapter {

    CustomAdapter(Context context, Cursor c, boolean autoRequery) {
      super(context, c, autoRequery);
    }

    @Override
    public View newView(Context context, Cursor cursor, ViewGroup viewGroup) {
      return new TextView(context);
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
      if (view instanceof TextView) {
        ((TextView) view).setText(cursor.getString(1));
      }
    }
  }

  private static class CustomSupportAdapter extends CursorAdapter {

    CustomSupportAdapter(Context context, Cursor c, boolean autoRequery) {
      super(context, c, autoRequery);
    }

    @Override
    public View newView(Context context, Cursor cursor, ViewGroup viewGroup) {
      return new TextView(context);
    }

    @Override
    public void bindView(View view, Context context, Cursor cursor) {
      if (view instanceof TextView) {
        ((TextView) view).setText(cursor.getString(1));
      }
    }
  }
}

File: sample/src/main/java/com/schibsted/spain/barista/sample/WrappedEditTextActivity.kt
package com.schibsted.spain.barista.sample

import android.content.Context
import android.database.Cursor
import android.database.MatrixCursor
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.view.ViewGroup
import android.widget.CursorAdapter
import android.widget.SearchView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.textfield.TextInputLayout

class WrappedEditTextActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_wrapped_edittext)
    val searchResult = findViewById<TextView>(R.id.searchResult)
    val searchView = findViewById<SearchView>(R.id.searchview)
    val supportSearchView = findViewById<androidx.appcompat.widget.SearchView>(R.id.supportSearchView)
    applyListener(searchResult, searchView)
    applySupportListener(searchResult, supportSearchView)
    applyAdapter(searchView)
    applySupportAdapter(supportSearchView)
    val textInputLayout = findViewById<TextInputLayout>(R.id.textInput)
    applyTextInputListener(searchResult, textInputLayout)
  }

  private fun applyListener(searchResult: TextView, searchView: SearchView) {
    searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
      override fun onQueryTextSubmit(query: String): Boolean {
        searchResult.text = query
        return true
      }

      override fun onQueryTextChange(newText: String): Boolean {
        searchResult.text = newText
        return true
      }
    })
  }

  private fun applySupportListener(searchResult: TextView, searchView: androidx.appcompat.widget.SearchView) {
    searchView.setOnQueryTextListener(object : androidx.appcompat.widget.SearchView.OnQueryTextListener {
      override fun onQueryTextSubmit(query: String): Boolean {
        searchResult.text = query
        return true
      }

      override fun onQueryTextChange(newText: String): Boolean {
        searchResult.text = newText
        return true
      }
    })
  }

  private fun applyTextInputListener(searchResult: TextView, textInputLayout: TextInputLayout) {
    val editText = textInputLayout.editText
    editText?.addTextChangedListener(object : TextWatcher {
      override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
      override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
        searchResult.text = charSequence
      }

      override fun afterTextChanged(editable: Editable) {}
    })
  }

  private fun applyAdapter(searchView: SearchView) {
    val cursor = generateCursor()
    val adapter: CursorAdapter = CustomAdapter(this, cursor, true)
    searchView.suggestionsAdapter = adapter
  }

  private fun applySupportAdapter(searchView: androidx.appcompat.widget.SearchView) {
    val cursor = generateCursor()
    val adapter: androidx.cursoradapter.widget.CursorAdapter = CustomSupportAdapter(this, cursor, true)
    searchView.suggestionsAdapter = adapter
  }

  private fun generateCursor(): MatrixCursor {
    val cursor = MatrixCursor(arrayOf("_id", "name"))
    cursor.addRow(arrayOf(0, "a"))
    cursor.addRow(arrayOf(1, "b"))
    return cursor
  }

  private class CustomAdapter internal constructor(context: Context?, c: Cursor?, autoRequery: Boolean) :
    CursorAdapter(context, c, autoRequery) {
    override fun newView(context: Context, cursor: Cursor, viewGroup: ViewGroup): View {
      return TextView(context)
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
      if (view is TextView) {
        view.text = cursor.getString(1)
      }
    }
  }

  private class CustomSupportAdapter internal constructor(context: Context?, c: Cursor?, autoRequery: Boolean) :
    androidx.cursoradapter.widget.CursorAdapter(context, c, autoRequery) {
    override fun newView(context: Context, cursor: Cursor, viewGroup: ViewGroup): View {
      return TextView(context)
    }

    override fun bindView(view: View, context: Context, cursor: Cursor) {
      if (view is TextView) {
        view.text = cursor.getString(1)
      }
    }
  }
}